cfe_internal/enterprise/federation/federation.cf

```diff
@@ -0,0 +1,210 @@
+body file control { inputs => { "$(sys.libdir)/stdlib.cf" }; }
+
+bundle agent federation_config
+{
+  vars:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_dir" string => "/opt/cfengine/federation";
+      "bin_dir" string => "$(federation_dir)/bin";
+      "path" string => "$(federation_dir)/cfapache/federation-config.json";
+  classes:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_enabled"
+        expression => fileexists("$(path)"),
+        scope => "namespace";
+  vars:
+    federation_enabled::
+      "data" data => readjson("$(path)");
```

Comment:
This can cause errors if the file doesnt exist (bundle will be entered and get pre-evaluated), so I usually guard reading data on file presence. Then again, it's a legitimate expectation that the file exist if the federation is enabled so it might be a desirable error condition to keep.

Reply:
The existence of this file decides whether federation is enabled or not. I am essentially doing what you're asking, but I chose the name `federation_enabled` instead of `config_file_exists`.